### DISCUSSION TITLE
stop UAC prompts

### DISCUSSION THREAD

**Discussion by rebootitltd** [Romanitho/Winget-AutoUpdate#771] [Q&A]
so when you run it in both system and user context, you get the popups for admin elevation if the user does not have admin rights. anyway to stop this ?

**Comment by KnifMelti** [comment]
UAC prompts is a result of the user not being admin if updating something in user context that has a dependency to something that must be installed in system context.
Make sure you deploy the dependency beforehand in system context.for all clients.

**Reply by rebootitltd** [reply]
Yep I totally understand that, the things its trying to update are like visual c++ packages and default things in windows, so this would be an issue for anyone deploying this with both system and user context. so solutions are what, add the current local user to admin group before hand and remove them after, elevate the powershell before running in user context ? 

the usecase here is using this awesome tool to takeover updates for as many apps as possible in both system and user context where end users are not local admins (as they should not be!!!) but with these prompts it does not work..

**Reply by KnifMelti** [reply]
Visual C++ packages are bad players in WinGet world...
...often, they leave remnants in the uninstall part in registry where WinGet looks up if apps are installed or not, resulting in an install loop (the old version or bit type is detected as installed all the time).

Whitelist (included_apps.txt) and only SYSTEM updates are what we use in WAU, curating the list/mods for our fleet of 6K clients/20K non-admin users/500+ managed applications (of them 100 exist in our whitelist).

- User installed apps update on their own (in an ordered user-friendly way) and more often than the WinGet Repo manages to stay updated.
- SYSTEM installed apps that can be updated on their own (service/task) are not in the whitelist
- Apps with updating problems (that can't be solved with mods) don't make it to the list

**Reply by rebootitltd** [reply]
ahh I see you control what it touches, thats smart I did not think of that, I was hoping to use this like a one fix all updater for my clients

I might experiment with using a block list for common UAC prompt offenders, see if I can get it honed down, windows updates should handle c++ and the other ones anyway

**Reply by KnifMelti** [reply]
There's no ring to rule them all...
